x/bank/keeper/send.go

```diff
@@ -254,13 +255,18 @@ func (k BaseSendKeeper) setBalance(ctx sdk.Context, addr sdk.AccAddress, balance
 	}
 
 	accountStore := k.getAccountStore(ctx, addr)
+	denomPrefixStore := k.getDenomPrefixStore(ctx, balance.Denom)
 
-	// Bank invariants require to not store zero balances.
+	// x/bank invariants prohibit persistence of zero balances
 	if balance.IsZero() {
 		accountStore.Delete([]byte(balance.Denom))
+		denomPrefixStore.Delete(address.MustLengthPrefix(addr))
 	} else {
 		bz := k.cdc.MustMarshal(&balance)
 		accountStore.Set([]byte(balance.Denom), bz)
+
+		// store a reverse index from denomination to account address
+		denomPrefixStore.Set(address.MustLengthPrefix(addr), bz)
```

Comment:
Should we store the balance `bz` in both stores? I guess that's fine, but maybe more efficient to just store `[]byte{1}`?

Reply:
We need to store something, so I stored balance. We can also store a sentinel value like `""` or `1`.

Reply:
Does `""` actually work? I always got errors when I tried that

Reply:
a byte of an empty string should work, I'll give it a shot.

Reply:
Probably best to not have redundant storage right?